### PR TITLE
[Feature] Support native reader for paimon table

### PR DESCRIPTION
**Pull request by miomiocat**
Read performance improved up to 15 times

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java [hunk 1]
```diff
@@ -132,7 +148,43 @@ public void setupScanRangeLocations(TupleDescriptor tupleDescriptor, ScalarOpera
         scanNodePredicates.setSelectedPartitionIds(selectedPartitions.values());
     }
 
-    private void addScanRangeLocations(DataSplit split, String predicateInfo) {
+    private THdfsFileFormat fromType(String type) {
+        THdfsFileFormat tHdfsFileFormat;
+        switch (type) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:153]: could you need to convert it to lower case?

Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:153]: There's no need, as the Paimon SDK will provide the format in lowercase directly.

**Review by stephen-shelby — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by miomiocat** [comment]
@mergify backport branch-3.2

**Comment by DorianZheng** [comment]
@mergify backport branch-3.1

**Pull request merged**